main/src/ca/uwaterloo/flix/language/errors/ResolutionError.scala

```diff
@@ -186,6 +186,26 @@ object ResolutionError {
     }
   }
 
+  /**
+    * Inaccessible Sig Error.
+    *
+    * @param sym the sig symbol.
+    * @param ns  the namespace where the symbol is not accessible.
+    * @param loc the location where the error occurred.
+    */
+  case class InaccessibleSig(sym: Symbol.SigSym, ns: Name.NName, loc: SourceLocation) extends ResolutionError {
+    def summary: String = "Inaccessible."
+    def message: VirtualTerminal = {
+      val vt = new VirtualTerminal
+      vt << Line(kind, source.format) << NewLine
+      vt << ">> Definition '" << Red(sym.toString) << s"' is not accessible from the namespace '" << Cyan(ns.toString) << "'." << NewLine
+      vt << NewLine
+      vt << Code(loc, "inaccessible definition.") << NewLine
+      vt << NewLine
+      vt << Underline("Tip:") << " Mark the definition as public." << NewLine
+    }
+  }
+
```

Comment:
So, what is the deal with visibility of classes?

Do we want both classes and their signatures to have visibility? I guess it could make sense.

Reply:
I'm not sure what the use-case of a private sig in a public class would be. And a public sig in a private class seems nonsensical. Maybe it would be better to keep visibility on the class only.